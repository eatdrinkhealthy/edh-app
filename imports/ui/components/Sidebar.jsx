// @flow
import React from "react";
import { Link } from "react-router-dom";

const Sidebar = (): React$Element<*> => (
  <div className="sidebar">
    <Link to="/">Home</Link>
    <h2 className="mh4">Eat Drink Healthy</h2>
    <div className="pitch">
      A platform to find, share, and discuss healthy places to shop and eat.
    </div>
    <div className="get-started">Get Started</div>
    <div className="pt2 pl4 pr4 pb4">
      <form className="measure center">
        <fieldset id="sign_up" className="ba b--transparent ph0 mh0">
          <legend className="f4 fw6 ph0 mh0">Sign In</legend>
          <div className="mt3">
            <label className="db fw6 lh-copy f6" htmlFor="email-address">Email</label>
            <input
              className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              type="email"
              name="email-address"
              id="email-address"
            />
          </div>
          <div className="mv3">
            <label className="db fw6 lh-copy f6" htmlFor="password">Password</label>
            <input
              className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
              type="password"
              name="password"
              id="password"
            />
          </div>
          <label className="pa0 ma0 lh-copy f6 pointer" htmlFor="remember">
            <input type="checkbox" id="remember" /> Remember me
          </label>
        </fieldset>
        <div className="">
          <input
            className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
            type="submit"
            value="Sign in"
          />
        </div>
        <div className="lh-copy mt3">
          <a href="#0" className="f6 link dim black db">Sign up</a>
          <a href="#0" className="f6 link dim black db">Forgot your password?</a>
        </div>
      </form>
    </div>
  </div>
);

export default Sidebar;
