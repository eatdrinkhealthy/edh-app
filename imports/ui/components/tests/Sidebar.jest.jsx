// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Sidebar from "../Sidebar";
import appReducer from "../../../state/reducers";

describe("<Sidebar />", function () {
  // Per a React blog post, when using renderer, must mock out refs
  // https://facebook.github.io/react/blog/2016/11/16/react-v15.4.0.html#mocking-refs-for-snapshot-testing
  // eslint-disable-next-line flowtype/no-weak-types
  function createNodeMock(element: HTMLElement): ?Object {
    if (element.type === "input") {
      return {
        focus() {},
      };
    }
    return null;
  }
  const options = { createNodeMock };

  const testStore = createStore(appReducer);

  it("matches render snapshot - no action form", function () {
    const tree = renderer.create(
      <MemoryRouter>
        <Sidebar />
      </MemoryRouter>,
      options,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - login form", function () {
    // TODO figure out why MemoryRouter isn't passing location to child
    const tree = renderer.create(
      <MemoryRouter initialEntries={["/sidebar?action=login"]} initialIndex={0}>
        <Sidebar location={{ pathname: "/sidebar", search: "?action=login", hash: "" }} />
      </MemoryRouter>,
      options,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - signup form", function () {
    // TODO figure out why MemoryRouter isn't passing location to child
    const tree = renderer.create(
      <Provider store={testStore}>
        <MemoryRouter initialEntries={["/sidebar?action=signup"]} initialIndex={0}>
          <Sidebar location={{ pathname: "/sidebar", search: "?action=signup", hash: "" }} />
        </MemoryRouter>
      </Provider>,
      options,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
