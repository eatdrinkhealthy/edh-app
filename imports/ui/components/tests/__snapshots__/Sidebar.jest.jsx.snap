// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> matches render snapshot - login form 1`] = `
<div
  className="sidebar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <h2
    className="mh4"
  >
    Eat Drink Healthy
  </h2>
  <div
    className="pitch"
  >
    A platform to find, share, and discuss healthy places to shop and eat.
  </div>
  <div
    className="get-started"
  >
    Get Started
  </div>
  <div
    className="pt2 pl4 pr4 pb4"
  >
    <form
      className="measure center"
      onSubmit={[Function]}
    >
      <fieldset
        className="ba b--transparent ph0 mh0"
        id="sign_in"
      >
        <legend
          className="f4 fw6 ph0 mh0"
        >
          Login
        </legend>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="usernameEmail"
          >
            Username / Email
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="usernameEmail"
            name="usernameEmail"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="mv3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </fieldset>
      <div>
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          type="submit"
          value="Login"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`<Sidebar /> matches render snapshot - no action form 1`] = `
<div
  className="sidebar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <h2
    className="mh4"
  >
    Eat Drink Healthy
  </h2>
  <div
    className="pitch"
  >
    A platform to find, share, and discuss healthy places to shop and eat.
  </div>
  <div
    className="get-started"
  >
    Get Started
  </div>
  <div />
</div>
`;

exports[`<Sidebar /> matches render snapshot - signup form 1`] = `
<div
  className="sidebar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <h2
    className="mh4"
  >
    Eat Drink Healthy
  </h2>
  <div
    className="pitch"
  >
    A platform to find, share, and discuss healthy places to shop and eat.
  </div>
  <div
    className="get-started"
  >
    Get Started
  </div>
  <div
    className="pt2 pl4 pr4 pb4"
  >
    <form
      className="measure center"
      onSubmit={[Function]}
    >
      <fieldset
        className="ba b--transparent ph0 mh0"
        id="sign_up"
      >
        <legend
          className="f4 fw6 ph0 mh0"
        >
          Create Account
        </legend>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="username"
            name="username"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="mt3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
        <div
          className="mv3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
        <div
          className="mv3"
        >
          <label
            className="db fw6 lh-copy f6"
            htmlFor="confirmPassword"
          >
            Confirm Password
          </label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            id="confirmPassword"
            name="confirmPassword"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </fieldset>
      <div>
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          type="submit"
          value="Create Account"
        />
      </div>
    </form>
  </div>
</div>
`;
